Cpp math hypot (#1114)
Codecademy/docs: content/cpp/concepts/math-functions/terms/hypot/hypot.md

---
Title: 'hypot()'

> Description: 'Returns the square root of sum of square of arguments passed.'

Reviewer: ```suggestion
Description: 'Returns the square root of sum of squares of arguments passed.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
  - 'Methods'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **hypot()** function returns the square root of sum of square of arguments passed.

Reviewer 2: ```suggestion
The **hypot()** function returns the value of the longest side of a right-angled triangle.
```

## Syntax

> ```cpp

Reviewer 2: ```suggestion
```pseudo
```

 hypot(float x, float y)
```

> It finds the hypotenuse, hypotenuse is the longest side of a right angled triangle.

Reviewer: ```suggestion
The `hypot()` function finds the hypotenuse, the longest side of a right angled triangle. The parameters `x` and `y` are the lengths of the other two sides.
```

> The `hypot()` function finds the hypotenuse, the longest side of a right angled triangle. The parameters `x` and `y` are the lengths of the other two sides.

Reviewer 2: ```suggestion
The `hypot()` function finds the hypotenuse, the longest side of a right-angled triangle. The parameters `x` and `y` are the floating-point lengths of the other two sides.
```

## Example

> The following example uses the `hypot()` function to find the hypotanuse of a right angled triangle with base and perpendicular as `9` & `10`:

Reviewer: ```suggestion
The following example uses the `hypot()` function to find the hypotenuse of a right triangle with base and perpendicular sides of `9` & `10`:
```

> The following example uses the `hypot()` function to find the hypotenuse of a right triangle with base and height of `9` & `10`:

Reviewer 2: ```suggestion
The following example uses the `hypot()` function to find the hypotenuse of a right triangle with `base` and `height` values:
```

> The following example uses the `hypot()` function to find the hypotenuse of a right triangle with base and perpendicular sides of `9` & `10`:

Reviewer: ```suggestion
The following example uses the `hypot()` function to find the hypotenuse of a right triangle with base and height of `9` & `10`:
```

```cpp
#include <cmath>
#include <iostream>
using namespace std;

> int main()
> {
>     double x = 9, y = 10, res;
>     res = hypot(x, y);
>   
>     cout << res << endl;
>     return 0;
>     This results in the following output: 13.4536
> }

Reviewer 2: ```suggestion

int main() {
  double base = 9, height = 10, res;
  res = hypot(base, height);
  
  cout << res << endl;
  return 0;
}
```

The last part didn't seem to fit right in the \```cpp code block.

```

> 

Reviewer 2: ```suggestion

This results in the following output:

```

Reviewer 2: Then I'd suggest adding a new \```shell snippet that looks like this:

```
13.4536
```

## Codebyte Example

> The following example is runnable and returns the square root of sum of squares of `4.525252` & `5.767676`:

Reviewer 2: ```suggestion
The following example is runnable and returns the square root of sum of squares of `4.525252` and `5.767676`:
```

```codebyte/cpp
#include <cmath>
#include <iostream>
using namespace std;

> int main()
> {
>     long double a, b, result;
>     a = 4.525252;
>     b = 5.767676;
>   
>     result = hypot(a, b);
>     cout << result;
>     return 0;
> }

Reviewer 2: ```suggestion

int main() {
  long double a, b, result;
  a = 4.525252;
  b = 5.767676;
  
  result = hypot(a, b);
  cout << result;
  return 0;
}
```

---
Title: 'hypot()'
Description: 'Returns the square root of sum of squares of arguments passed.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
  - 'Methods'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **hypot()** function returns the value of the longest side of a right-angled triangle.

Reviewer 3: ```suggestion
The **`hypot()`** function returns the value of the longest side of a right-angled triangle.
```

## Syntax

```pseudo

>  hypot(float x, float y)

Reviewer 3: ```suggestion
hypot(float x, float y)
```

```

The `hypot()` function finds the hypotenuse, the longest side of a right-angled triangle. The parameters `x` and `y` are the floating-point lengths of the other two sides.

## Example

The following example uses the `hypot()` function to find the hypotenuse of a right triangle with `base` and `height` values:

```cpp
#include <cmath>
#include <iostream>
using namespace std;

> int main()

Reviewer 3: ```suggestion

int main()
```

{

>     double x = 9, y = 10, res;
>     res = hypot(x, y);
>   
>     cout << res << endl;
>     return 0;

Reviewer 3: ```suggestion
  double x = 9, y = 10, res;
  res = hypot(x, y);
  
  cout << res << endl;
  return 0;
```

}
```

This results in the following output:

```shell
13.4536
```

## Codebyte Example

The following example is runnable and returns the square root of sum of squares of `4.525252` and `5.767676`:

```codebyte/cpp
#include <cmath>
#include <iostream>
using namespace std;

> int main() {

Reviewer 3: ```suggestion
int main()
{
```
